File: e2e/playwright-typescript/tests/create-task/create-task.spec.ts
import { test } from '@/fixtures';
import { TaskContext } from '@/context/TaskContext';
import * as allure from "allure-js-commons";
import { AllureEpic } from '@/data/AllureEpic';

test.beforeAll(async () => {
  await allure.epic(AllureEpic.TASK_MANAGEMENT);
  await allure.feature('Create task');
});

test.describe('Create task', () => {
  let context: TaskContext;

  test.beforeEach(async ({ support }) => {
    context = new TaskContext();
    const response = context.createTaskResponse();

    await support.mock.api.getTasks([response]);
    await support.mock.api.createTask(response);
    await support.mock.api.getTask(response.id, response);
    await support.mock.api.getIsValid(response.id, true);
  });

  test('should create task', async ({ step, validate }) => {
    // given
    await step.navigation.openMainPage();

    // when
    await step.tasks.openCreateTaskForm();
    await step.tasks.createTask.fillForm(context.createTaskData());
    await step.tasks.createTask.submitForm();

    // then
    await validate.tasks.hasTask(context.title);

    // when
    await step.tasks.openTaskInfoForm(context.title);

    // then
    await validate.task.data(context.createTaskData());
  });
});

File: e2e/capybara-ruby/spec/create_task/create_task_spec.rb
# frozen_string_literal: true

require 'spec_helper'

require_relative '../../context/task_context'
require_relative '../../data/allure_epic'

RSpec.describe 'Create task', type: :system,
                             epic: AllureEpic::TASK_MANAGEMENT,
                             feature: 'Create task',
                             tms: '100' do
  let(:context) { TaskContext.new }

  before do
    response = context.create_task_response
    support.mock.api
         .get_tasks([response])
         .create_task(response)
         .get_task(response[:id], response)
         .get_is_valid(response[:id], true)
  end

  it 'should create task' do
    # given
    step.navigation.open_main_page

    # when
    step.tasks.open_create_task_form
         .fill_form(context.create_task_data)
         .submit_form

    # then
    validate.tasks.has_task(context.title)

    # when
    step.tasks.open_task_info_form(context.title)

    # then
    validate.task.data(context.create_task_data)
  end
end
